drivers/hmis/app/models/hmis/form/form_processor.rb

```diff
@@ -208,13 +208,17 @@ def exit_factory(create: true)
   def ce_assessment_factory(create: true)
     return owner if owner.is_a? Hmis::Hud::Assessment
 
-    self.ce_assessment ||= enrollment_factory.assessments.build(user_id: custom_assessment&.user_id) if create
+    return ce_assessment if ce_assessment.present? || !create
```

Comment:
```suggestion
    return ce_assessment if !create
```

I think this would be a little clearer.

* If we are _not_ creating a new record, than we just return what we have.
* If we _are_ creating a new record, then we return what we have OR we build a new one. Thats what the conditional assignment operator `||=` does below. (It only executes the right-side if the the left-side is not present.

Reply:
Although, now I see you are following the pattern from a bunch of other factories. Maybe then it is better to leave what you have but replace the `||=` with a `=`. Lets go with that for consistency!